src/Tribe/Cache.php

```diff
@@ -536,4 +536,93 @@ public function data_size_over_packet_size( $value ) {
 		// If the size of the string is above 90% of the database `max_allowed_packet` setting, then it should not be written to the db.
 		return $size > ( $feature_detection->get_mysql_max_packet_size() * .9 );
 	}
+
+	/**
+	 * Returns a transient that might have been stored, due ot its size, in chunks.
+	 *
+	 * @since TBD
+	 *
+	 * @param string               $id                 The name of the transients to return.
+	 * @param string|array<string> $expiration_trigger The transient expiration trigger(s).
+	 *
+	 * @return false|mixed Either the transient value, joined back into one, or `false` to indicate
+	 *                     the transient was not found or was malformed.
+	 */
+	public function get_chunkable_transient( $id, $expiration_trigger = '' ) {
+		$transient = $this->get_id( $id, $expiration_trigger );
+
+		if ( wp_using_ext_object_cache() ) {
+			return get_transient( $transient );
+		}
+
+		$chunks = [];
+		$i      = 0;
+		do {
+			$chunk_transient            = $transient . '_' . $i ++;
+			$chunk                      = get_transient( $chunk_transient );
+			$chunks[ $chunk_transient ] = (string) $chunk;
+		} while ( ! empty( $chunk ) );
+
+		if ( count( array_filter( $chunks ) ) === 0 ) {
+			return false;
+		}
+
+		try {
+			$data          = implode( '', $chunks );
+			$is_serialized = preg_match( '/^[aO]:\\d+:/', $data );
+			$unserialized  = maybe_unserialize( implode( '', $chunks ) );
+			if ( is_string( $unserialized ) && $unserialized === $data && $is_serialized ) {
+				// Something was messed up.
+				return false;
+			}
+		} catch ( Exception $e ) {
+			return false;
+		}
+	}
+
+	/**
+	 * Sets a transient in the database with the knowledge that, if too large to be stored in one
+	 * DB row, it will be chunked.
+	 *
+	 * The method will redirect to the `set_transient` function if the site is using object caching.
+	 *
+	 *
+	 * @since TBD
+	 *
+	 * @param string               $id                 The transient ID.
+	 * @param mixed                $value              The value to store, that could be chunked.
+	 * @param int                  $expiration         The transient expiration, in seconds.
+	 * @param string|array<string> $expiration_trigger The transient expiration trigger(s).
+	 *
+	 * @return bool Whether the transient, or the transient chunks, have been stored correctly or not.
+	 */
+	public function set_chunkable_transient( $id, $value, $expiration = 0, $expiration_trigger = '' ) {
+		$transient = $this->get_id( $id, $expiration_trigger );
+
+		if ( wp_using_ext_object_cache() ) {
+			return $this->set_transient( $transient, $value, $expiration );
+		}
+
+		$inserted         = [];
+		$serialized_value = maybe_serialize( $value );
+		$chunk_size       = tribe( 'feature-detection' )->get_mysql_max_packet_size() * .9;
```

Comment:
Just to make it more clear. 
```suggestion
		$chunk_size       = tribe( 'feature-detection' )->get_mysql_max_packet_size() * 0.9;
```